src/hooks/useSearchArticleLanguage.js

```diff
@@ -0,0 +1,59 @@
+import { useState, useEffect, useLayoutEffect } from 'preact/hooks'
+import { getLanglinks } from 'api'
+
+export const useSearchArticleLanguage = (lang, title) => {
+  const [items, setItems] = useState([])
+  const [query, setQuery] = useState()
+  const [allLanguages, setAllLanguages] = useState([])
+
+  useEffect(() => {
+    getLanglinks(lang, title)
+      .then(items => {
+        const languages = items.map(item => {
```

Comment:
This map could be in the API transform function to do it only once and cache it.

Reply:
good point! on it now.